airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -127,6 +127,9 @@ definitions:
     properties:
       name:
         type: string
+        description: Stream's name (deprecated, will be replaced by stream_name).
+      stream_name:
```

Comment:
would it be simpler if instead of adding this `AirbyteStreamName` object, you just added namespace as a property to this top level object? then we don't need to deprecate `name`? it also saves us having to ever have name and stream_name at the same time.

Reply:
i got to think it makes the backwards compatibility story way simpler too.

Reply:
in my last PR,  @michel-tricot was mentioning this: https://github.com/airbytehq/airbyte/pull/1993#discussion_r572588441

Did I misunderstand his comment?

Reply:
kk. i don't think you misunderstood.